cudet/updates.py

```diff
@@ -0,0 +1,116 @@
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+import subprocess
+import yaml
+import glob
+from copy import deepcopy
+from fuelclient.commands import base
+
+MU_UPGRADE_DICT_RESTRICTED = {
+    'mu_upgrade': {
+        'enabled': True,
+        'restart_rabbit': True,
+        'restart_mysql': True,
+    }
+}
+MU_UPGRADE_DICT_FULL = deepcopy(MU_UPGRADE_DICT_RESTRICTED)
+MU_UPGRADE_DICT_FULL['mu_upgrade'].update({
+    'metadata': {
+        'group': 'general',
+        'label': 'MU label',
+        'restrictions': [
+            {'action': 'hide'}
+        ],
+        'weight': 10
+    },
+    'type': 'hidden',
+    'value': True
+})
+
+
+def shell_cmd(cmdlist, stdin=None, stdout=None, stderr=subprocess.PIPE,
+              ignore_exceptions=False):
+    try:
+        proc = subprocess.Popen(
+            cmdlist,
+            stdin=stdin,
+            stdout=stdout,
+            stderr=stderr,
+            bufsize=1
+        )
+        proc.wait()
+        result = {}
+        if proc.returncode > 0:
+            message = "Error: " + proc.stderr.read() if proc.stderr \
+                else "command returned {0}, something went wrong.".format(
+                    proc.returncode)
+            raise Exception(message)
+        if stdin:
+            result["stdin"] = proc.stdin
+        if stdout:
+            result["stdout"] = proc.stdout
+        if stderr:
+            result["stderr"] = proc.stderr
+    except Exception as e:
+        if not ignore_exceptions:
+            raise e
+    return result
+
+
+class Updates(base.BaseCommand):
+    """TODO: provide meaningful description"""
+
+    def get_parser(self, prog_name):
+        parser = super(Updates, self).get_parser(prog_name)
+        parser.add_argument('env_id',
+                            type=str,
+                            help='ID of the environment')
+        parser.add_argument('install',
+                            type=bool, dest='do_install'
+                            help='Install MU')
+        return parser
+
+    def _update_settings_file(filename, full=True):
+        settings_yaml = None
+        with open(filename, 'r') as f:
+            settings_yaml = yaml.load(f.read())
+        settings_yaml['editable'].update(
+            MU_UPGRADE_DICT_FULL if full else MU_UPGRADE_DICT_RESTRICTED)
+        with open(filename, 'w') as f:
+            stream = yaml.dump(settings_yaml, default_flow_style=False)
+            f.write(stream.replace('{', ''))
+
+    def take_action(self, parsed_args):
+        if parsed_args.do_install:
+            try:
+                env_id = parsed_args.env_id
+                result = shell_cmd(['fuel', 'deployment', '--download', '--env', env_id])
```

Comment:
Every call to the "fuel" cli command should also contain "--os-username" & "--os-password" arguments, because if user changed fuel's password the command will fail and there is no way to avoid that. Moreover, I propose to use internal fuelclient objects instead of calling external command because they already contain an authentication mechanism.